src/components/ha-hls-player.ts

```diff
@@ -0,0 +1,193 @@
+import {
+  css,
+  CSSResult,
+  customElement,
+  html,
+  LitElement,
+  property,
+  PropertyValues,
+  query,
+  TemplateResult,
+} from "lit-element";
+import { fireEvent } from "../common/dom/fire_event";
+import { nextRender } from "../common/util/render-status";
+import { getExternalConfig } from "../external_app/external_config";
+import type { HomeAssistant } from "../types";
+
+type HLSModule = typeof import("hls.js");
+
+@customElement("ha-hls-player")
```

Comment:
Shouldn't we also use this in `ha-camera-stream`?

Reply:
tbh I didn't want to fuck it up

Reply:
This is just a copy/paste right?

Reply:
Completed. Twas easy once I could test it :)

Probably could use more testing